map/src/map/layers/FavoriteLayer.js

```diff
@@ -163,7 +163,10 @@ const FavoriteLayer = () => {
                         return;
                     }
                     layer.options.isFavorite = true;
-                    layer.on('click', onClick);
+                    if (!layer.options.hasClickHandler) {
+                        layer.on('click', onClick);
+                        layer.options.hasClickHandler = true;
+                    }
```

Comment:
Extra `layer.options.hasClickHandler` seems redundand.
It should be better to try useCallback instead of "function onClick()".